src/videojs-http-streaming.js

```diff
@@ -471,7 +502,8 @@ class HlsHandler extends Component {
     }
     this.setOptions_();
     // add master playlist controller options
-    this.options_.url = this.source_.src;
+    this.options_.src = (this.source_.src && (/^data:/i).test(this.source_.src)) ?
```

Comment:
seems like this can be simplified to just `expandDataUri(this.source_.src)` since it returns the string argument if it isn't a data uri.